eclipse-collections-code-generator/src/main/resources/impl/map/mutable/objectPrimitiveHashMap.stg

```diff
@@ -512,6 +512,20 @@ public class Object<name>HashMap\<K> implements MutableObject<name>Map\<K>, Exte
         return value;
     }
 
+    @Override
+    public <type> getAndPut(K key, <type> putValue, <type> defaultValue)
+    {
+        int index = this.probe(key);
+        if (isNonSentinel(this.keys[index]) && nullSafeEquals(this.toNonSentinel(this.keys[index]), key))
+        {
+            <type> existingValue = this.values[index];
+            this.addKeyValueAtIndex(key, putValue, index);
```

Comment:
Why is this not just `this.values[index] = putValue;`?

Reply:
Changed.